src/datasets/io/spark.py

```diff
@@ -0,0 +1,40 @@
+from typing import Optional
+
+import pyspark
+
+from .. import Features, NamedSplit
+from ..download import DownloadMode
+from ..packaged_modules.spark.spark import Spark
+from .abc import AbstractDatasetReader
+
+
+class SparkDatasetReader(AbstractDatasetReader):
+    def __init__(
+        self,
+        df: pyspark.sql.DataFrame,
+        split: Optional[NamedSplit] = None,
+        features: Optional[Features] = None,
+        cache_dir: str = None,
+        force_download: bool = False,
+        **kwargs,
+    ):
+        super().__init__(
+            path_or_paths="unused",
```

Comment:
I guess you can modify `AbstractDatasetReader`'s typing to allow `None` here
```suggestion
            path_or_paths=None,
```

Reply:
Done